EVALUATION_STANDARDS.md

```diff
@@ -0,0 +1,224 @@
+    
+*   **Ethical and Safety Standards**
+    
+    *   **Safety Checks:**
+        *   Avoid designing evaluations that encourage unsafe or unethical behavior.
+        *   For risky capabilities, use simulations or subtasks to limit real-world implications.
+    *   **Bias Mitigation:**
+        *   Regularly review datasets, prompts, and scoring mechanisms for biases.
+    
+*   **Inspect Repository Standards**
+    
+    *   **Template Compliance:**
+        *   Follow the repository structure and include required files like `README.md`, `CONTRIBUTING.md`, and `METADATA.json`.
+        *   Pass all automated checks in the template repository.
+    *   **Version Control:**
+        *   Pin all dependencies and document all software versions for reproducibility.
+            *   For Python dependencies used by the evaluation itself (i.e. not inside the sandbox), document the dependencies in `pyproject.toml` using [uv projects](https://docs.astral.sh/uv/guides/projects/).
+    
+*   **Testing and Debugging**
+    
+    *   **Unit Testing:**
+        *   Ensure each custom tool, solver, and scorer has dedicated unit tests.
+    *   **End-to-End Testing:**
+        *   Use mocked LLM APIs for comprehensive evaluation testing without incurring high costs.
+    *   **QA Logs Review:**
+        *   Read logs to confirm the absence of errors and correct solver behavior.
+    
+*   **Documentation**
+    
+    *   **README.md**
+        
+        *   Ensure the README contains instructions for setting up the evaluation. It should be possible, by reading the README, to mechanically follow instructions from a freshly-cloned repo to running the evaluation.
+            
+    *   **CONTRIBUTING.md**
+        
+        *   Dependency maintenance should be documented here: where are dependency versions located? Where should they be updated?
+            
+        *   Tests should be documented here: where are tests located? How can I run the tests? What different kinds of tests are there?
+            
+        *   Any other maintenance tasks should be documented here, e.g. any external APIs, any cached datasets or tools that might need to be updated, etc.
+            
+    *   **Code**
+        
+        *   All code should be either self-documenting (i.e. the name of the variable, function or class is self-explanatory) or explicitly documented with docstrings and comments.
+            
+        *   All code should be type-annotated.
+            
+        *   Type annotations should be automatically verified by running `mypy` .
+            
+
+## **Best Practices Summary**
+
+*   **Clarity and Consistency**
+    
+    *   Define clear objectives for each evaluation, with specific pass/fail criteria.
+    *   Use consistent system prompts and instructions across tasks to ensure fair comparisons.
+    *   Document all templates, datasets, and assumptions for transparency and reproducibility.
+    
+*   **Statistical Rigor**
+    
+    *   TODO
```

Comment:
@alexandraabbas any suggestions for what to add here for v1?

Reply:
Some suggestions:
* Naive vs clustered mean: Your dataset might be skewed towards certain question types/tasks. Cluster standard errors on the unit of [randomization](https://en.wikipedia.org/wiki/Randomization) (for example, passage of text).
* Epochs: Use Inspect's built-in epoch parameter to resampling answers from the same model several times, and using the question-level averages as the question scores.
* On top of average scores, report the standard error of the mean.

See more details [here](https://www.anthropic.com/research/statistical-approach-to-model-evals)